### PR TITLE
#160609549 generate jwt token

### PR DESCRIPTION
**Pull request by MuhanguziDavid**
### What does this PR do?

Creates a JWT token on signup and login

### Description of tasks to be completed?

Currently, users do not receive an authentication token when they register an account

This task adds a JWT token to be received by users when the register and when they login such that they can be able to use system features.

### How should this be manually tested?

Get the deployment link on heroku (Pasted below for convenience.)
https://ah-backend-targaryen-sta-pr-14.herokuapp.com
Open postman, paste the link, and add relevant login url `/api/users/login/`. Provide the username, email, and password in the body in the format stated below.
```source-json
{
	"user": {
		"email": "davidm@jake.jake",
		"password": "1234567890"
	}
}
```
You should be able to receive a JWT token on successful registration.

### What are the relevant pivotal tracker stories?

[#160609549](https://www.pivotaltracker.com/story/show/160609549)

### Any background context you want to add?

N/A

### Screenshots
screen shot 2018-10-01 at 10 41 01

### CODE REVIEW

File: authors/apps/authentication/tests.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        self.user_missing_name = {'user':{'username':None, 'email':'bridget@gmail.com', 'password':'password_bridget'}}
+
+        self.response = self.client.post('/api/users/',self.user_data,format="json")
+        
+    def test_api_can_create_a_user(self):
+        """Test the api has user creation capability."""
+        self.assertEqual(self.response.status_code, status.HTTP_201_CREATED)
+    
+    def test_register_with_an_existing_email(self):
+        """Tests that user will not be cfreated with an exisiting email"""
```
Comment (RonKbS) [inline authors/apps/authentication/tests.py]: There is a typo here @MuhanguziDavid

**Review by kakaemma — APPROVED** [review]
Good work @MuhanguziDavid. However, kindly look into how it should manually be tested and give more details.

File: authors/apps/authentication/tests.py (hunk 1 above)
Comment (MuhanguziDavid) [reply authors/apps/authentication/tests.py]: Thanks @RonKbS, let me correct that

**Review by RonKbS — APPROVED** [review]
nice work @MuhanguziDavid.

**Comment by MuhanguziDavid** [comment]
I have given more details on how the PR should be manually tested

**Comment by lym** [comment]
👍

**Pull request merged**